File: test/language/templates/java/templates/InstantiateOnlyNestedTest.java
package templates;

import static org.junit.jupiter.api.Assertions.*;
import org.junit.jupiter.api.Test;

import java.io.IOException;

import zserio.runtime.io.BitStreamReader;
import zserio.runtime.io.ByteArrayBitStreamReader;
import zserio.runtime.io.ByteArrayBitStreamWriter;

import templates.instantiate_only_nested.InstantiateOnlyNested;
import templates.instantiate_only_nested.N32;
import templates.instantiate_only_nested.pkg.Test_uint32;

public class InstantiateOnlyNestedTest
{
    @Test
    public void readWrite() throws IOException
    {
        final InstantiateOnlyNested instantiateOnlyNested = new InstantiateOnlyNested();
        instantiateOnlyNested.setTest32(new Test_uint32(new N32(13)));

        final ByteArrayBitStreamWriter writer = new ByteArrayBitStreamWriter();
        instantiateOnlyNested.write(writer);

        final BitStreamReader reader = new ByteArrayBitStreamReader(
                writer.toByteArray(), writer.getBitPosition());
        final InstantiateOnlyNested readInstantiateOnlyNested = new InstantiateOnlyNested(reader);
        assertTrue(instantiateOnlyNested.equals(readInstantiateOnlyNested));
    }

}

File: test/language/templates/cpp98/InstantiateOnlyNestedTest.cpp
#include "gtest/gtest.h"

#include "templates/instantiate_only_nested/InstantiateOnlyNested.h"

namespace templates
{
namespace instantiate_only_nested
{

TEST(InstantiateOnlyNestedTest, readWrite)
{
    InstantiateOnlyNested instantiateOnlyNested;
    N32 n32;
    n32.setValue(13);
    pkg::Test_uint32 t32;
    t32.setValue(n32);
    instantiateOnlyNested.setTest32(t32);

    zserio::BitStreamWriter writer;
    instantiateOnlyNested.write(writer);
    size_t bufferSize = 0;
    const uint8_t* buffer = writer.getWriteBuffer(bufferSize);

    zserio::BitStreamReader reader(buffer, bufferSize);
    InstantiateOnlyNested readInstantiateOnlyNested(reader);

    ASSERT_TRUE(instantiateOnlyNested == readInstantiateOnlyNested);
}

} // namespace instantiate_only_nested
} // namespace templates
